group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -445,6 +446,42 @@ public List<ListGroupsResponseData.ListedGroup> listGroups(List<String> statesFi
         return groupStream.map(group -> group.asListedGroup(committedOffset)).collect(Collectors.toList());
     }
 
+
+    public List<ConsumerGroupDescribeResponseData.DescribedGroup> consumerGroupDescribe(
+        List<String> groupIds,
+        long committedOffset
+    ) {
+        List<ConsumerGroupDescribeResponseData.DescribedGroup> response = new ArrayList<>();
+
+        for (String groupId: groupIds) {
+            Group group = groups.get(groupId, committedOffset);
```

Comment:
I wonder if we could follow a bit more what we did in `describeGroups`. Would it be possible?

For instance, there we use `genericGroup` which handles the group type check as well. We could use a similar method for consumer groups, I think.